backend/lcfs/web/api/notification/repo.py

```diff
@@ -76,7 +91,86 @@ async def get_notification_messages_by_user(
 
         # Execute the query and retrieve the results
         result = await self.db.execute(query)
-        return result.scalars().all()
+        return result.unique().scalars().all()
+
+    def _apply_notification_filters(
+        self, pagination: PaginationRequestSchema, conditions: List
+    ):
+        for filter in pagination.filters:
+            filter_value = filter.filter
+            filter_option = filter.type
+            filter_type = filter.filter_type
+
+            # Handle date filters
+            if filter.field == "date":
+                filter_value = filter.date_from
+                field = get_field_for_filter(NotificationMessage, 'create_date')
+            elif filter.field == 'user':
+                field = get_field_for_filter(NotificationMessage, 'related_user_profile.first_name')
+            else:
+                field = get_field_for_filter(NotificationMessage, filter.field)
+            conditions.append(
+                apply_filter_conditions(
+                    field, filter_value, filter_option, filter_type
+                )
+            )
+
+        return conditions
+
+    @repo_handler
+    async def get_paginated_notification_messages(
+        self, user_id, pagination: PaginationRequestSchema
+    ) -> tuple[Sequence[NotificationMessage], int]:
+        """
+        Queries notification messages from the database with optional filters. Supports pagination and sorting.
+
+        Args:
+            pagination (dict): Pagination and sorting parameters.
+
+        Returns:
+            List[NotificationSchema]: A list of notification messages matching the query.
+        """
+        conditions = [NotificationMessage.related_user_profile_id == user_id]
+        pagination = validate_pagination(pagination)
+
+        if pagination.filters:
+            self._apply_notification_filters(pagination, conditions)
+
+        offset = 0 if (pagination.page < 1) else (pagination.page - 1) * pagination.size
+        limit = pagination.size
+        # Start building the query
+        query = (
+            select(NotificationMessage)
+            .options(
+                joinedload(NotificationMessage.related_organization),
+                joinedload(NotificationMessage.origin_user_profile)
+                .joinedload(UserProfile.user_roles)
+                .joinedload(UserRole.role),
+            )
+            .where(and_(*conditions))
+        )
+
+        # Apply sorting
+        if not pagination.sort_orders:
+            query = query.order_by(NotificationMessage.create_date.desc())
+        # for order in pagination.sort_orders:
```

Comment:
Do we need this commented out code?

Reply:
thanks, missed to add sorting logic here. updated code.